Client/Application/LeanplumIntegration.swift

```diff
@@ -0,0 +1,196 @@
+            Leanplum.setAppId(settings.appId, withProductionKey: settings.key)
+        }
+        Leanplum.syncResourcesAsync(true)
+        setupTemplateDictionary()
+
+        var userAttributesDict = [AnyHashable: Any]()
+        userAttributesDict["Focus Installed"] = "false"
+        userAttributesDict["Klar Installed"] = "false"
+        userAttributesDict["Alternate Mail Client Installed"] = "mailto:"
+
+        if let focusURL = URL(string: "firefox-focus://"), UIApplication.shared.canOpenURL(focusURL) {
+            userAttributesDict["Focus Installed"] = "true"
+        }
+
+        if let klarURL = URL(string: "firefox-klar://"), UIApplication.shared.canOpenURL(klarURL) {
+            userAttributesDict["Klar Installed"] = "true"
+        }
+
+        if let mailtoScheme = self.profile?.prefs.stringForKey(PrefsKeys.KeyMailToOption), mailtoScheme != "mailto:" {
+            userAttributesDict["Alternate Mail Client Installed"] = mailtoScheme
+        }
+
+        Leanplum.start(userAttributes: userAttributesDict)
+
+        Leanplum.track(LeanplumEventName.openedApp.rawValue)
+    }
+
+    // Events
+
+    func track(eventName: LeanplumEventName) {
+        if shouldSendToLP() {
+            Leanplum.track(eventName.rawValue)
+        }
+    }
+
+    func track(eventName: LeanplumEventName, withParameters parameters: [String: AnyObject]) {
+        if shouldSendToLP() {
+            Leanplum.track(eventName.rawValue, withParameters: parameters)
+        }
+    }
+
+    // States
+
+    func advanceTo(state: String) {
+        if shouldSendToLP() {
+            Leanplum.advance(to: state)
+        }
+    }
+
+    // Data Modeling
+
+    func setupTemplateDictionary() {
+        if shouldSendToLP() {
+            LPVar.define("Template Dictionary", with: ["Template Text": "", "Button Text": "", "Deep Link": "", "Hex Color String": ""])
+        }
+    }
+
+    func getTemplateDictionary() -> [String:String]? {
+        if shouldSendToLP() {
+            return Leanplum.object(forKeyPathComponents: ["Template Dictionary"]) as? [String : String]
+        }
+        return nil
+    }
+
+    func getBoolVariableFromServer(key: String) -> Bool? {
+        if shouldSendToLP() {
+            return Leanplum.object(forKeyPathComponents: [key]) as? Bool
+        }
+        return nil
+    }
+
+    // Utils
+    
+    func setEnabled(_ enabled: Bool) {
+        self.enabled = enabled
+        if self.enabled {
+            self.start()
+        }
+    }
+
+    func shouldShowFocusUI() -> Bool {
+        guard let shouldShowFocusUI = LeanplumIntegration.sharedInstance.getBoolVariableFromServer(key: "shouldShowFocusUI"), let focus = URL(string: "firefox-focus://"), let klar = URL(string: "firefox-klar://"), !UIApplication.shared.canOpenURL(focus) && !UIApplication.shared.canOpenURL(klar) && shouldShowFocusUI else {
+            return false
+        }
+        return true
+    }
```

Comment:
Caution: bugs hide in long lines.

The logic here is unclear. Suggest re-writing to be clearer, without the `guard` (you're returning either way).

`guard` is neat because it eliminates some exceptional cases, but they can be difficult to read because they flip the control flow.

We're already in the `LeanplumIntegration` singleton, so we can just call the instance method.

Trying to parse the logic here, I've written it as
```swift
let shouldShowFocusUI = getBoolVariableFromServer(key: "shouldShowFocusUI") ?? false
let otherBrowsers = supportsScheme("firefox-klar") || supportsScheme("firefox-klar")
return shouldShowFocusUI && !otherBrowsers
```

I'm still not sure I have the intent of `shouldShowFocusUI()`; you may want to check `shouldShowFocusUI` (the variable before you calculate `otherBrowsers`.